File: src/main/java/org/jukeboxmc/network/handler/EmoteListHandler.java
package org.jukeboxmc.network.handler;

import org.jukeboxmc.Server;
import org.jukeboxmc.network.packet.EmoteListPacket;
import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class EmoteListHandler implements PacketHandler<EmoteListPacket> {

    @Override
    public void handle( EmoteListPacket packet, Server server, Player player ) {
        player.getEmotes().addAll( packet.getEmotes() );
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/network/handler/EmoteListHandler.kt
package org.jukeboxmc.server.network.handler

import org.cloudburstmc.protocol.bedrock.packet.EmoteListPacket
import org.jukeboxmc.server.JukeboxServer
import org.jukeboxmc.server.player.JukeboxPlayer

class EmoteListHandler : PacketHandler<EmoteListPacket> {

    override fun handle(packet: EmoteListPacket, server: JukeboxServer, player: JukeboxPlayer) {
        player.setEmotes(packet.pieceIds)
    }
}